src/app/dataset/preview/preview.component.ts

```diff
@@ -248,58 +264,78 @@ export class PreviewComponent extends SubscriptionManager implements OnInit, OnD
   *  - get and show samples based on plugin
   *  - loads historyVersions based on plugin
   */
-  getXMLSamples(plugin: PluginType): void {
-    let loadingSamples = true;
-    let loadingHistories = true;
+  getXMLSamples(plugin: PluginType, prefilling?: boolean): void {
+    this.isLoadingHistories = true;
+    this.isLoadingSamples = true;
 
-    this.isLoading = true;
-    this.allSampleComparisons = [];
-    this.selectedComparison = undefined;
+    if (!prefilling) {
+      this.allSampleComparisons = [];
+      this.previewFilters.comparison = undefined;
+    }
 
     this.onClickedOutside();
     this.editorConfig = this.editorPrefs.getEditorConfig(true);
-    this.selectedPlugin = plugin;
-    this.previewFilters.pluginType = plugin;
+    this.previewFilters.basic.pluginType = plugin;
     this.setPreviewFilters.emit(this.previewFilters);
+
     this.subs.push(
-      this.workflows.getWorkflowSamples(this.filteredExecutionId, plugin).subscribe((result) => {
-        this.allSamples = this.undoNewLines(result);
+      this.workflows.getWorkflowSamples(this.filteredExecutionId, plugin).subscribe(
+        (result) => {
+          this.allSamples = this.undoNewLines(result);
 
-        if (this.allSamples.length === 1) {
-          this.expandedSample = 0;
-        }
-        loadingSamples = false;
-        if (!loadingHistories) {
-          this.isLoading = false;
+          if (this.allSamples.length === 1) {
+            this.expandedSample = 0;
+          }
+
+          this.sampleRecordIds = [];
+          this.allSamples.forEach((sample) => {
+            this.sampleRecordIds.push(sample.ecloudId);
+          });
+
+          const prvCmp = this.previewFilters.comparison;
+
+          if (prefilling && prvCmp && prvCmp.pluginType && prvCmp.executionId) {
+            const comparisonPluginType = prvCmp.pluginType;
+            const comparisonExecutionId = prvCmp.executionId;
+            this.getXMLSamplesCompare(comparisonPluginType, comparisonExecutionId, prefilling);
+            this.isLoadingSamples = false;
+          } else {
+            this.isLoadingSamples = false;
+          }
```

Comment:
First of all, the `isLoadingSamples` assignment can be extracted from both branches.

Second: I feel the code would be a lot nicer if the call to `getXMLSamplesCompare` is taken out of this method. It is only triggered by the call from `previewFilters()` (line 374 below) and it would mean this simplifies the code here enormously. Also, this way we don't have to do Subscription within Subscription. But the question is: is there a time dependency here?

Reply:
On your first point I concur.  On your second, I chained the call to avoid a race condition.  In the course of variable tidying the execution flow may now allow that - I'll check....